packages/status-react/src/protocol/use-sorted-chats.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from 'react'
+
+import { useProtocol } from './provider'
+
+import type { Community } from '@status-im/js'
+
+export type Chat = Community['description']['chats'][0] & {
+  id: string
+}
+
+export type Category = {
+  id: string
+  name: string
+  position: number
+  chats: Chat[]
+}
+
+type Result = {
+  categories: Category[]
+  chats: Chat[]
+}
+
+function sortByPosition<T extends { position: number }>(items: T[]): T[] {
+  items.sort((a, b) => {
+    if (a.position < b.position) {
+      return -1
+    }
+    if (a.position > b.position) {
+      return 1
+    }
+    return 0
+  })
+
+  return items
+}
+
+export const useSortedChats = (): Result => {
+  const { community } = useProtocol()
+
+  return useMemo<Result>(() => {
+    const categoryChats: Record<string, Chat[]> = {}
+
+    const chats = Object.entries(community.chats).reduce<Chat[]>(
+      (acc, [chatId, chat]) => {
+        const parsedChat: Chat = {
+          id: chatId,
+          ...chat,
+        }
+
+        if (chat.categoryId && community.categories[chat.categoryId]) {
+          categoryChats[chat.categoryId] ??= []
+          categoryChats[chat.categoryId].push(parsedChat)
+        } else {
+          acc.push(parsedChat)
+        }
+
+        return acc
+      },
+      []
+    )
+
+    const categories = Object.entries(categoryChats).map(([id, chats]) => ({
+      ...community.categories[id],
+      id,
+      chats: sortByPosition(chats),
+    }))
+
+    return {
+      categories: sortByPosition(categories),
+      chats: sortByPosition(chats),
+    }
```

Comment:
non-blocking: would you prefer if we did all the mapping and sorting in a `Community.getCategories` method in `status-js` instead?

Reply:
or even in `getChats` method or `chats` getter